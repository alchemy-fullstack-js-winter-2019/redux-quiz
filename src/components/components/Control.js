import React from 'react';
import PropTypes from 'prop-types';

function Control({ title, onChange, subtitle }) {
  return (
    <>
    <label> Title </label>
    <input name="title" value={title} type="text" onChange={onChange} />
    <label> Subtitle </label>
    <input name="subtitle" value={subtitle} type="text" onChange={onChange} />
    </>
  );
}

Control.propTypes =  {
  title: PropTypes.string,
  onChange: PropTypes.func,
  subtitle: PropTypes.string
};

export default Control;
