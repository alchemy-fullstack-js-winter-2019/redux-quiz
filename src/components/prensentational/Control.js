import React from 'react';
import PropTypes from 'prop-types';

function Control({ title, onChange, subTitle }) {
  return (
    <>
    <label> Title </label>
    <input name="title" value={title} type="text" onChange={onChange} />
    <label> Subtitle </label>
    <input name="subTitle" value={subTitle} type="text" onChange={onChange} />
    </>
  );
}

Control.propTypes =  {
  title: PropTypes.string.isRequired,
  onChange: PropTypes.func.isRequired,
  subTitle: PropTypes.string.isRequired
};

export default Control;
