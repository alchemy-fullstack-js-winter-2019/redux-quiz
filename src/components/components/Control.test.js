import React from 'react';
import renderer from 'react-test-renderer';
import Control from './Control';

describe('Character', () => {
  it('matches a snapshot', () => {
    const title = 'title';

    const subtitle = 'subtitle';

    const tree = renderer.create(
      <Control title={title} subtitle={subtitle}/>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
