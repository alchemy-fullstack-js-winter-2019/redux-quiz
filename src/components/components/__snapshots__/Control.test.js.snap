// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Character matches a snapshot 1`] = `
Array [
  <label>
     Title 
  </label>,
  <input
    name="title"
    type="text"
    value="title"
  />,
  <label>
     Subtitle 
  </label>,
  <input
    name="subtitle"
    type="text"
    value="subtitle"
  />,
]
`;
